### PR TITLE
[refactor] Refactored agent with main and help function #33

### LINKED ISSUE #15: [bug] get_neighbors() may fail with "Interrupted system call"
```
lua: /usr/sbin/netjson-monitoring:67: Interrupted system call
stack traceback:
	[C]: in function '(for generator)'
	/usr/sbin/netjson-monitoring:67: in function 'get_ip_neigh'
	/usr/sbin/netjson-monitoring:84: in function 'get_neighbors'
	/usr/sbin/netjson-monitoring:181: in main chunk
	[C]: ?
```

I need to test this on more devices before ruling out if it's a system issue or something that has to be dealt with in the code.

### LINKED ISSUE #33: [refactor] Refactor script with a main and help function
This change is requested here:-
> Can you please refactoring the script that it has more have a look and feel of a C programm with a main and help function?
> Have a look at https://github.com/openwrt/packages/blob/master/net/mwan3/files/usr/sbin/mwan3rtmon thats only an example.
> From my point of view, this has certain advantages. If you later decide to make a C program out of it, then you already know what you need for functions. My experience as a maintainer has often shown me that.
> 
> Also add a valid license header

https://github.com/openwisp/openwrt-openwisp-monitoring/pull/25#discussion_r637421072

### PR DESCRIPTION
**Pull request by devkapilbansal**
Closes #33

### CODE REVIEW

**Comment by devkapilbansal** [comment]
Converting this to draft as it will result in merge conflicts with #53

**Review by nemesisdesign — COMMENTED** [review]
@feckert WDYT?

File: openwrt-openwisp-monitoring/files/monitoring.init [hunk 1]
```diff
@@ -69,7 +69,6 @@ start_service() {
 		exit 1
 	fi
 	interval="--interval $interval"
-	monitored_interfaces="--monitored_interfaces \"$monitored_interfaces\""
```
Comment (devkapilbansal) [inline openwrt-openwisp-monitoring/files/monitoring.init:72]: This was left by mistake. Removed now

File: openwrt-openwisp-monitoring/files/monitoring.agent [hunk 2]
```diff
@@ -2,72 +2,35 @@
...
+	printf "General options:\n"
+	printf "  --help\t\t\t\t: Show this help text\n"
+	printf "  -v, --version\t\t\t\t: Shows version of the agent.\n"
+	printf "  --mode <send, collect>\t\t: Mode for running Monitoring Agent.\n"
+	printf "Openwisp config options:\n"
+	printf "  --url <server-url>\t\t\t: URL of the OpenWISP server.\n"
+	printf "  --uuid <uuid>\t\t\t\t: UUID of the device.\n"
+	printf "  --key <key>\t\t\t\t: Key for the device.\n"
+	printf "  --verify_ssl <0, 1>\t\t\t: Whether SSL Authentication should be enabled or not.\n"
+	printf "Openwisp Monitoring config options:\n"
```
Comment (nemesisdesign) [inline openwrt-openwisp-monitoring/files/monitoring.agent:20]: Openwisp > OpenWISP here and above

File: openwrt-openwisp-monitoring/files/monitoring.init (hunk 1 above)
Comment (nemesisdesign) [reply openwrt-openwisp-monitoring/files/monitoring.init:72]: Thanks!

File: openwrt-openwisp-monitoring/files/monitoring.agent [hunk 3]
```diff
@@ -219,17 +182,83 @@ send_data() {
...
+				exit 1
+				;;
+			*) break ;;
+		esac
+		shift
+	done
+
+	INTERVAL=${INTERVAL:-300}
+	VERBOSE_MODE=${VERBOSE_MODE:-0}
+	TMP_DIR="/tmp/openwisp/monitoring"
```
Comment (feckert) [inline openwrt-openwisp-monitoring/files/monitoring.agent:246]: Should we not create this dir with `mkdir -p` ?

Comment (devkapilbansal) [reply openwrt-openwisp-monitoring/files/monitoring.agent:246]: I am doing in *save_data* function
https://github.com/openwisp/openwrt-openwisp-monitoring/blob/7be1724500399ecc0393e4e3fb9ba3fce879c4ec/openwrt-openwisp-monitoring/files/monitoring.agent#L128-L129

**Review by nemesisdesign — APPROVED** [review]

**Pull request merged**